### PR TITLE
General improvements and addition of GenericService and GenericRepository.

### PR DESCRIPTION
**Pull request by JamesMcFall**
Adding lots of iterative changes that have been made to Nodebrick to the project create template such as config changes, code tweaks and tests.

Notes: Tests pass but coverage is not yet at 100%.

### CODE REVIEW

File: .docker/docker-compose.yml [hunk 1]
```diff
@@ -21,6 +21,13 @@ services:
       - PROJECT_ID
       - MODULE_ID
       - NODEBRICK_CONFIG_PATH
+      - NODEBRICK_API_APP_PORT
+      - NODEBRICK_DATABASE_POSTGRES_HOST
```
Comment (nolazybits) [inline .docker/docker-compose.yml]: you don't want those here as they are module specific.
Add them to the compose of the modules requiring them

File: packages/nodebrick-cli/assets/templates/project/business-module/{{moduleParamCase}}/src/repositories/IGenericRepository.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        else
+        {
+            throw new ErrorDBEntityNotFound(entity.constructor, entity.uuid);
+        }
+    }
+
+    public async findOne(id: number): Promise<TModel | undefined>
+    {
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        const entityType: TModel = this._metadata.target as any;
```
Comment (nolazybits) [inline packages/nodebrick-cli/assets/templates/project/business-module/{{moduleParamCase}}/src/repositories/IGenericRepository.ts]: why as any if know the type is TModel?

File: packages/nodebrick-cli/assets/templates/project/business-module/{{moduleParamCase}}/src/services/IGenericService.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+        return this._repository.update(model);
+    }
+
+    public async deleteByUUID(uuid: UUID): Promise<TModel>
+    {
+        return this._repository.deleteOneByUUID(uuid);
+    }
+
+    // @todo Add other common methods (like find by UUID) etc into this generic service.
```
Comment (nolazybits) [inline packages/nodebrick-cli/assets/templates/project/business-module/{{moduleParamCase}}/src/services/IGenericService.ts]: still valid?

Comment (JamesMcFall) [reply packages/nodebrick-cli/assets/templates/project/business-module/{{moduleParamCase}}/src/services/IGenericService.ts]: Oops, no. Removed

File: .docker/docker-compose.yml (hunk 1 above)
Comment (JamesMcFall) [reply .docker/docker-compose.yml]: I don't understand what you mean - this mirrors current projects. Both of those config options are required for the project that is created to work. So it's appropriate for them to be in the .docker/docker-compose.yml file no?

Or are you saying shift the nodebrick database ones to nodebrick-database/.docker/docker-compose?

File: packages/nodebrick-cli/assets/templates/project/business-module/{{moduleParamCase}}/src/repositories/IGenericRepository.ts (hunk 2 above)
Comment (JamesMcFall) [reply packages/nodebrick-cli/assets/templates/project/business-module/{{moduleParamCase}}/src/repositories/IGenericRepository.ts]: Messaged on Slack.

**Pull request merged**